### PR TITLE
Feature: Add Steam Guard support

### PR DESCRIPTION
**Pull request by CorySanin**
Addresses #24

Adds Steam Guard as an option. I came from Android where my favorite TOTP apps all had Steam support built-in. Probably missed some things (~~tests are on their way, unless this PR is a non-starter~~ (created TOTP tests)). Cut me some slack, I've never touched Swift before!

What I've manually tested and verified:

- [x] Correct generation of Steam Guard tokens
- [x] Export/import in 2FAS format v4
- [x] Import from Andotp
- [x] Import from Aegis
- [x] Scanning QR codes from Andotp and Aegis
- [x] Scanning QR codes from KeePassXC
- [x] Manual secret input
- [x] Tokens display correctly in (square) widget
- [x] Unit tests run and pass

> if I recall, there are problems with exporting their secret to any other app.

You're right, it's a pain in the neck. I successfully exported mine from my rooted Android phone, but there's a more accessible solution. Create a new secret with [steamguard-cli](https://github.com/dyc3/steamguard-cli).

### CODE REVIEW

**Review by adocyn — COMMENTED** [review]
Small changes required, but before any merge we'll have to check if:

- current file changes are compatible with 5.2.0 where there was a large refactor
- if editing/adding manually still works
- what about sync, as it's the most difficult part

File: TwoFAS/TwoFAS/Other/Generated/T.generated.swift [hunk 1]
```diff
@@ -1175,10 +1175,10 @@ internal enum T {
   internal enum Tokens {
     /// Point your camera at the screen to capture the QR code.
     internal static let addDescription = T.tr("Localizable", "tokens__add_description", fallback: "Point your camera at the screen to capture the QR code.")
-    /// Enter secret key
-    internal static let addEnterManual = T.tr("Localizable", "tokens__add_enter_manual", fallback: "Enter secret key")
-    /// Upload an image with the QR code
-    internal static let addFromGallery = T.tr("Localizable", "tokens__add_from_gallery", fallback: "Upload an image with the QR code")
+    /// Enter secret key manually
```
[the remark this answers is not in the document]
Comment (CorySanin) [reply TwoFAS/TwoFAS/Other/Generated/T.generated.swift:1178]: In that case I'll revert the change to this file. I thought swiftgen was unintentionally skipping generation of this one.

[the remark this answers is not in the document]
Comment (CorySanin) [reply TwoFAS/TwoFAS/Other/Generated/T.generated.swift:1178]: Or are you saying neither generated file needs to be touched?

**Comment by adocyn** [comment]
I've checked the task board and we can try to add it to 5.2.5 for testing purposes.

File: TwoFAS/TwoFAS/Root/Modules/AddingService/Token/Interactor/AddingServiceTokenModuleInteractor.swift [hunk 2]
```diff
@@ -98,6 +100,8 @@ extension AddingServiceTokenModuleInteractor: AddingServiceTokenModuleInteractin
     
     func clear() {
         switch serviceTokenType {
+        case .steam:
```
[the remark this answers is not in the document]
Comment (CorySanin) [reply TwoFAS/TwoFAS/Root/Modules/AddingService/Token/Interactor/AddingServiceTokenModuleInteractor.swift:103]: Ah, with a tuple. Got it.

**Comment by CorySanin** [comment]
Added tests for TOTP token generation

**Pull request merged**